hetio/hetnet.py

```diff
@@ -293,8 +301,36 @@ def extract_metapaths(self, source_kind, target_kind, max_length):
                     current_metapaths.append(new_metapath)
             metapaths.extend(current_metapaths)
             previous_metapaths = current_metapaths
-        metapaths = [metapath for metapath in metapaths
-                     if metapath.target() == target]
+
+        if target:
+            metapaths = [metapath for metapath in metapaths
+                         if metapath.target() == target]
+        return metapaths
+
+    def extract_all_metapaths(self, max_length, exclude_inverts=False):
+        """
+        Extract all metapaths up to max_length possible by walking the
+        metagraph. Unlike extract_metapaths, this function is does not limit
+        to a specific source metanode. Unlike metaedges/edges, metapaths do not
+        have standard orientations. However, it is possible for a metapath to
+        be an inverse of another metapath, e.g. CbGaD and DaGbC.
+
+        Set exclude_inverts=True to return only one orientation of two inverse
+        metapaths. This may be useful if metapaths are being used to compute
+        symmetric metrics, such as path count or DWPC. In this case, you may
+        want to optimize by computing values for only one metapath orientation.
+        """
+        metapaths = list()
+        metapaths_with_inverses = set()
+        metanodes = self.get_nodes()
+        for source in metanodes:
+            from_source = self.extract_metapaths(source, max_length=max_length)
+            for metapath in from_source:
+                if exclude_inverts and metapath in metapaths_with_inverses:
+                    continue
+                metapaths.append(metapath)
+                for covered in metapath, metapath.inverse:
```

Comment:
Minor comment, but is there a different word than `covered` we could use here? Maybe `covered_metapath` or some synonym of covered? No big issue to just keep it though, either way.

Reply:
Will think of something.